catalog/confluence/catalog-confluence-source/src/main/java/org/codice/ddf/confluence/source/ConfluenceSource.java

```diff
@@ -0,0 +1,386 @@
+
+    private Date lastAvailableDate = null;
+
+    private long availabilityPollInterval = TimeUnit.SECONDS.toMillis(60);
+
+    private Set<SourceMonitor> sourceMonitors = new HashSet<>();
+
+    private Map<String, Set<String>> additionalMetacardAttributes = new HashMap<>();
+
+    public ConfluenceSource(FilterAdapter adapter, EncryptionService encryptionService,
+            ConfluenceInputTransformer transformer, ResourceReader reader) {
+        this.filterAdapter = adapter;
+        this.encryptionService = encryptionService;
+        this.transformer = transformer;
+        this.resourceReader = reader;
+    }
+
+    public void init() {
+        if (StringUtils.isNotBlank(username) && StringUtils.isNotBlank(password)) {
+            factory = new SecureCxfClientFactory<SearchResource>(endpointUrl,
+                    SearchResource.class,
+                    username,
+                    password);
+        } else {
+            factory = new SecureCxfClientFactory<SearchResource>(endpointUrl, SearchResource.class);
+        }
+    }
+
+    @Override
+    public String getConfigurationPid() {
+        return configurationPid;
+    }
+
+    @Override
+    public void setConfigurationPid(String configurationPid) {
+        this.configurationPid = configurationPid;
+    }
+
+    @Override
+    public boolean isAvailable() {
+        boolean isAvailable = false;
+        if (!lastAvailable || (lastAvailableDate.before(new Date(
+                System.currentTimeMillis() - availabilityPollInterval)))) {
+
+            Response response = null;
+
+            try {
+                response = factory.getWebClient()
+                        .head();
+            } catch (Exception e) {
+                LOGGER.debug("Web Client was unable to connect to endpoint.", e);
+                isAvailable = false;
+            }
+
+            if (response != null && !(response.getStatus() >= HttpStatus.SC_NOT_FOUND
+                    || response.getStatus() == HttpStatus.SC_BAD_REQUEST
+                    || response.getStatus() == HttpStatus.SC_PAYMENT_REQUIRED)) {
+                isAvailable = true;
+                lastAvailableDate = new Date();
+            }
+        } else {
+            isAvailable = lastAvailable;
+        }
+
+        if (lastAvailable != isAvailable) {
+            for (SourceMonitor monitor : this.sourceMonitors) {
+                if (isAvailable) {
+                    monitor.setAvailable();
+                } else {
+                    monitor.setUnavailable();
+                }
+            }
+        }
+
+        lastAvailable = isAvailable;
+        return isAvailable;
+    }
+
+    @Override
+    public boolean isAvailable(SourceMonitor callback) {
+        sourceMonitors.add(callback);
+        return isAvailable();
+    }
+
+    @Override
+    public SourceResponse query(QueryRequest request) throws UnsupportedQueryException {
+        SearchResource confluence = factory.getClient();
```

Comment:
You might want to consider moving the client get after you are sure the query request is valid.